### DISCUSSION TITLE
Rate-limiting issue updates

### DISCUSSION THREAD

**Discussion by DenverCoder1** [DenverCoder1/github-readme-streak-stats#180] [Announcements]
You may be coming from seeing your stats card show a rate-limiting error.

We are working to fix this issue. Leave a comment below if you have any questions or suggestions.

![error card](https://user-images.githubusercontent.com/20955511/136065796-d924a917-38e9-4438-b533-d3f4dc9323cc.png)

For the time being, you can try avoiding this error by deploying Streak Stats on your own.

Deploying to Vercel is free and it only takes a few steps to set up.

### Step-by-step instructions for deploying to Heroku

Heroku costs around $5-$7/month minimum for a single app.
  
1. Sign in to **Heroku** or create a new account at <https://heroku.com>
2. Visit [this link](https://github.com/settings/tokens/new?description=GitHub%20Readme%20Streak%20Stats) to create a new Personal Access Token (no scopes required)
3. Scroll to the bottom and click **"Generate token"**
4. Click the Deploy button below

[![](https://user-images.githubusercontent.com/20955511/136058102-b79570bc-4912-4369-b664-064a0ada8588.png)](#) [![Deploy to Heroku](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy?template=https://github.com/DenverCoder1/github-readme-streak-stats/tree/main "Deploy to Heroku")

5. **Add the token** as a Config Var with the key `TOKEN`:

![heroku config variables](https://user-images.githubusercontent.com/20955511/136292022-a8d9b3b5-d7d8-4a5e-a049-8d23b51ce9d7.png)

6. Click **"Deploy App"** at the end of the form
7. Once the app is deployed, you can use `<your-app-name>.herokuapp.com` in place of `streak-stats.demolab.com`

### Instructions for Deploying to Vercel (Free)

Vercel is a free hosting service that can be used to run PHP. **Note:** PNG mode is not supported since Inkscape will not be installed.

### Step-by-step instructions for deploying to Vercel

1. Sign in to **Vercel** or create a new account at <https://vercel.com>
2. Clone this repository with `git clone https://github.com/DenverCoder1/github-readme-streak-stats.git`
   - You may also fork the repository (make sure to include all branches) and clone your fork instead if you intend to make changes
3. Enter the directory with `cd github-readme-streak-stats`
4. Switch branches to the `vercel` branch with `git checkout vercel`
5. Make sure you have the [Vercel CLI](https://vercel.com/download) installed
6. Run `vercel` and follow the prompts to link your Vercel account and select a project name
7. The app will be deployed to `<project-name>.vercel.app`
8. Visit [this link](https://github.com/settings/tokens/new?description=GitHub%20Readme%20Streak%20Stats) to create a new Personal Access Token (no scopes required)
9. Scroll to the bottom and click **"Generate token"**
10. Visit the [Vercel dashboard](https://vercel.com/dashboard) and select your project, then click **"Settings"**, then **"Environment Variables"**.
11. Add a new variable with the key `TOKEN` and the value as your token from step 9 and click "Save".

![image](https://user-images.githubusercontent.com/20955511/209588756-8bf5b0cd-9aa6-41e8-909c-97bf41e525b3.png)

> **Note**  
> To set up automatic Vercel deployments from GitHub, make sure to turn **off** "Include source files outside of the Root Directory" in the General settings and use `vercel` as the production branch in the Git settings.

**Comment by DenverCoder1** [comment]
Requests have been updated to use GitHub's GraphQL API, so hopefully this should fix the issue.

**Update: This allows us to have more requests per hour but it is still not sufficient to prevent rate limiting entirely.**

**Comment by FleetAdmiralJakob** [comment]
Sorry, but self-hosting does not work right now on vercel. See #642 and #641
